File: java-integration/src/main/java/br/com/acqio/enums/TenantPlanMode.java
package br.com.acqio.enums;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public enum TenantPlanMode {

    DISABLED("Desativar"),
    TENANT("Lojista e Portador"),
    MDR_0("MDR-0 CRÉDITO/DEBITO");

    private final String description;

}

File: csharp/csharp/enums/TenantPlanMode.cs
namespace csharp.enums
{
    enum TenantPlanMode
    {
        DISABLED,
        TENANT,
        MDR_0
    }

    static class TenantPlanModeExtension
    {
        internal static string GetDescription(TenantPlanMode tenantPlanMode)
        {
            switch (tenantPlanMode)
            {
                case TenantPlanMode.MDR_0:
                    return "MDR-0 CRÉDITO/DEBITO";
                case TenantPlanMode.TENANT:
                    return "Lojista e Portador";
                default:
                    return "Desativar";
            }
        }

        internal static TenantPlanMode GetTenantPlanMode(string tenantPlanMode)
        {
            switch (tenantPlanMode)
            {
                case "MDR-0 CRÉDITO/DEBITO":
                    return TenantPlanMode.MDR_0;
                case "Lojista e Portador":
                    return TenantPlanMode.TENANT;
                default:
                    return TenantPlanMode.DISABLED;
            }
        }
    }
}
